src/thumb/load_addr.rs

```diff
@@ -0,0 +1,68 @@
+use num_derive::FromPrimitive;    
+use num_traits::FromPrimitive;
+use crate::types::Register;
+use crate::{get_bits, get_bit};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, FromPrimitive)]
+enum Source {
+    PC = 0,
+    SP = 1
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+struct LoadAddress { 
+    src: Source,
+    dest: Register,
+    word8: u8
```

Comment:
I think you should rename src -> address, word8 -> offset. Src makes me think it's the same as a source register, and offset is more descriptive here.